test/test_utils.py

```diff
@@ -268,6 +268,20 @@ def run_fn(tensor1, tensor2):
         out = checkpoint(run_fn, input_var, None)
         out.sum().backward()
 
+    def test_checkpoint_partial_grad(self):
+        def run_fn(tensor1, tensor2):
+            # tensor 2 is used for other application logic
+            return tensor1
+        input_var = torch.randn(1, 100, requires_grad=True)
```

Comment:
nit: please use a smaller size here as there is no need for such a big Tensor. Size of 2 or 4 would be enough.